addons/pos_adyen/static/src/js/payment_adyen.js

```diff
@@ -210,9 +210,13 @@ var PaymentAdyen = PaymentInterface.extend({
             // this promise don't resolve -- that is, it doesn't go to the 'then' clause.
             return Promise.reject(data);
         }).then(function (status) {
+            var line = self.pending_adyen_line();
+
+            if (line === undefined)
+                resolve(false);
+
             var notification = status.latest_response;
             var order = self.pos.get_order();
-            var line = self.pending_adyen_line();
 
```

Comment:
A big diff you can do in one line:
`!line && resolve(false);`

Please follow guidelines for commit messages:
https://www.odoo.com/documentation/16.0/contributing/development/coding_guidelines.html#commit-message-full-description